Add method to return signed URL and headers

waynelee opened the pull request:
Adds an API that returns just the URL and signed headers without sending the request to the server.

coderhaoxin commented on index.js:
```diff
@@ -52,6 +52,36 @@ OSS.prototype.generateSign = function(req, resource) {
+  headers['Authorization'] = this.generateSign(req, getResource(options));
+
+  // Generate URL
+  var url = require('url').format({
```
put `require('url')` to the top?

coderhaoxin commented on index.js:
```diff
@@ -52,6 +52,36 @@ OSS.prototype.generateSign = function(req, resource) {
+ */
+OSS.prototype.getSignedUrl = function(options) {
+  var headers = {};
+  Object.assign(headers, options.headers);
```
```js
var headers = Object.assign({}, options.headers);
```

coderhaoxin commented:
@waynelee add a test case for this?

coderhaoxin commented:
@waynelee Do you or someone else want to maintain this repo? I'll add a collaborator

waynelee commented:
@coderhaoxin. Updated new commit that addresses your comments. Also included a few extra changes to make the test work in my environment. We won't be doing much with Aliyun, it's just some exploratory work right now, so I don't think I'll be a good candidate to be a maintainer. I'd be happy to be a contributor if you'd like. Thanks for this module, it was very helpful!

coderhaoxin commented:
landed 92679471ed4fc33ccd24c6fa3cd1a38722130bd2, published as `v1.0.0`

coderhaoxin commented:
@waynelee collaborator added.
